Add CI with build and linting

kisialiou opened the pull request:
- Added simple CI with Clippy.
- Made minor changes to address points underlined by clippy.

karim-en reviewed (commented):
The contract isn't compilable anymore, please run `RUSTFLAGS='-C link-arg=-s' cargo build --target wasm32-unknown-unknown --release`

kisialiou commented:
> The contract isn't compilable anymore, please run `RUSTFLAGS='-C link-arg=-s' cargo build --target wasm32-unknown-unknown --release`

I fixed this and added an additional job which checks that the project is built with these flags and under the target.

karim-en commented on near/omni-prover/wormhole-omni-prover-proxy/src/lib.rs:
```diff
@@ -31,11 +31,12 @@ impl WormholeOmniProverProxy {
             .unwrap_or_else(|_| env::panic_str("ErrorOnArgsParsing"));
 
-        env::log_str(&format!("{}", args.vaa));
+        env::log_str(&args.vaa.to_string());
```
It is already string, so I would prefer to use `.clone()`

karim-en replied:
```suggestion
        env::log_str(&args.vaa);
```

karim-en commented on near/omni-types/src/evm/events.rs:
```diff
@@ -40,12 +40,12 @@ sol! {
     <V as TryFromLog<Log<T>>>::Error: std::fmt::Display,
 {
-    let rlp_decoded = Log::decode(&mut log_rlp.as_slice()).map_err(stringify)?;
+    let rlp_decoded = Log::decode(&mut log_rlp.to_vec().as_slice()).map_err(stringify)?;
```
And extra clone has been introduced here due to `to_vec()`, the `log_rlp` should be moved instead

kisialiou replied:
Are you sure? Initially, `log_rlp` was passed by value. Now, it's passed by immutable reference. Is moving it with mutable reference safe if it was not expected earlier?

karim-en replied:
Yes, it looks like Clippy didn't understand the flow correctly. the `log_rlp` should be moved as a `vec` because we need to modify it in the `Log::decode`

kisialiou replied:
I would simply return `Vec` here then

karim-en commented on near/omni-prover/evm-prover/src/lib.rs:
```diff
@@ -164,17 +171,17 @@ impl EvmProver {
+                } else {
                     Self::_verify_trie_proof(
-                        new_expected_root,
+                        new_expected_root.clone(),
```
extra clone

karim-en commented on near/omni-prover/evm-prover/src/lib.rs:
```diff
@@ -210,7 +217,7 @@ impl EvmProver {
                 let new_expected_root = get_vec(&node, 1);
                 Self::_verify_trie_proof(
-                    new_expected_root,
+                    new_expected_root.clone(),
```
extra clone

karim-en commented on near/omni-prover/evm-prover/src/lib.rs:
```diff
@@ -76,12 +81,13 @@ impl EvmProver {
                     .verify_proof_callback(
                         args.proof_kind,
-                        evm_proof.log_entry_data,
+                        evm_proof.log_entry_data.clone(),
```
extra clone
